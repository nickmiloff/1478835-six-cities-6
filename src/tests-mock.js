export const simpleMock = `test`;

export const cards = {
  changedCard: {
    id: 2,
    isFavorite: true
  },
  start: [
    {
      id: 1,
      isFavorite: true
    },
    {
      id: 2,
      isFavorite: false
    },
    {
      id: 3,
      isFavorite: false
    }
  ],
  end: [
    {
      id: 1,
      isFavorite: true
    },
    {
      id: 2,
      isFavorite: true
    },
    {
      id: 3,
      isFavorite: false
    }
  ],
  endWithRemove: [
    {
      id: 1,
      isFavorite: true
    },
    {
      id: 3,
      isFavorite: false
    }
  ]
};

export const auth = {
  api: {
    "id": 1,
    "email": `mail@mail.com`,
    "name": `name`,
    "avatar_url": `avatar.jpeg`,
    "is_pro": true
  },
  adapt: {
    id: 1,
    email: `mail@mail.com`,
    name: `name`,
    avatarUrl: `avatar.jpeg`,
    isPro: true
  }
};

export const user = {
  email: `mail@mail.com`,
  password: 12345678
};

export const cardsFull = {
  api: [
    {
      "city": {
        "name": `Paris`,
        "location": {
          "latitude": 48.85661,
          "longitude": 2.351499,
          "zoom": 13
        }
      },
      "images": [],
      "title": `Tile House`,
      "rating": 2,
      "type": `house`,
      "bedrooms": 2,
      "price": 161,
      "goods": [`Laptop friendly workspace`],
      "host": {
        "id": 25,
        "name": `Angelina`,
        "avatar_url": `img/avatar-angelina.jpg`,
        "is_pro": true
      },
      "description": `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
      "location": {
        "latitude": 48.85761,
        "longitude": 2.358499,
        "zoom": 16
      },
      "id": 1,
      "is_favorite": false,
      "is_premium": true,
      "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
      "max_adults": 9
    },
    {
      "city": {
        "name": `Amsterdam`,
        "location": {
          "latitude": 52.37454,
          "longitude": 4.897976,
          "zoom": 13
        }
      },
      "images": [],
      "title": `Loft Studio in the Central Area`,
      "rating": 3.6,
      "type": `apartment`,
      "bedrooms": 3,
      "price": 271,
      "goods": [
        `Laptop friendly workspace`,
        `Washer`,
        `Breakfast`,
        `Air conditioning`
      ],
      "host": {
        "id": 25,
        "name": `Angelina`,
        "avatar_url": `img/avatar-angelina.jpg`,
        "is_pro": true
      },
      "description": `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
      "location": {
        "latitude": 52.388540000000006,
        "longitude": 4.899976,
        "zoom": 16
      },
      "id": 2,
      "is_favorite": false,
      "is_premium": false,
      "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
      "max_adults": 8
    },
    {
      "city": {
        "name": `Paris`,
        "location": {
          "latitude": 48.85661,
          "longitude": 2.351499,
          "zoom": 13
        }
      },
      "images": [],
      "title": `Tile House`,
      "rating": 4.6,
      "type": `house`,
      "bedrooms": 2,
      "price": 865,
      "goods": [`Laptop friendly workspace`, `Breakfast`],
      "host": {
        "id": 25,
        "name": `Angelina`,
        "avatar_url": `img/avatar-angelina.jpg`,
        "is_pro": true
      },
      "description": `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
      "location": {
        "latitude": 48.83861,
        "longitude": 2.350499,
        "zoom": 16
      },
      "id": 3,
      "is_favorite": true,
      "is_premium": false,
      "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
      "max_adults": 8
    }
  ],
  adapted: [
    {
      city: {
        name: `Paris`,
        location: {
          latitude: 48.85661,
          longitude: 2.351499,
          zoom: 13,
        },
      },
      images: [],
      title: `Tile House`,
      rating: 2,
      type: `house`,
      bedrooms: 2,
      price: 161,
      goods: [`Laptop friendly workspace`],
      host: {
        id: 25,
        name: `Angelina`,
        avatarUrl: `img/avatar-angelina.jpg`,
        isPro: true,
      },
      description: `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.`,
      location: {
        latitude: 48.85761,
        longitude: 2.358499,
        zoom: 16,
      },
      id: 1,
      isFavorite: false,
      isPremium: true,
      previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg`,
      maxAdults: 9,
    },
    {
      city: {
        name: `Amsterdam`,
        location: {
          latitude: 52.37454,
          longitude: 4.897976,
          zoom: 13,
        },
      },
      images: [],
      title: `Loft Studio in the Central Area`,
      rating: 3.6,
      type: `apartment`,
      bedrooms: 3,
      price: 271,
      goods: [
        `Laptop friendly workspace`,
        `Washer`,
        `Breakfast`,
        `Air conditioning`,
      ],
      host: {
        id: 25,
        name: `Angelina`,
        avatarUrl: `img/avatar-angelina.jpg`,
        isPro: true,
      },
      description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
      location: {
        latitude: 52.388540000000006,
        longitude: 4.899976,
        zoom: 16,
      },
      id: 2,
      isFavorite: false,
      isPremium: false,
      previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
      maxAdults: 8,
    },
    {
      city: {
        name: `Paris`,
        location: {
          latitude: 48.85661,
          longitude: 2.351499,
          zoom: 13,
        },
      },
      images: [],
      title: `Tile House`,
      rating: 4.6,
      type: `house`,
      bedrooms: 2,
      price: 865,
      goods: [`Laptop friendly workspace`, `Breakfast`],
      host: {
        id: 25,
        name: `Angelina`,
        avatarUrl: `img/avatar-angelina.jpg`,
        isPro: true,
      },
      description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
      location: {
        latitude: 48.83861,
        longitude: 2.350499,
        zoom: 16,
      },
      id: 3,
      isFavorite: true,
      isPremium: false,
      previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/6.jpg`,
      maxAdults: 8,
    },
  ]
};

export const comments = {
  api: [
    {
      "comment": `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
      "date": `2019-05-08T14:13:56.569Z`,
      "id": 1,
      "rating": 4,
      "user": {
        "avatar_url": `img/1.png`,
        "id": 4,
        "is_pro": false,
        "name": `Max`
      }
    },
    {
      "comment": `A quiet of Amsterdam.`,
      "date": `2019-06-08T14:13:56.569Z`,
      "id": 2,
      "rating": 3,
      "user": {
        "avatar_url": `img/2.png`,
        "id": 5,
        "is_pro": false,
        "name": `Vadim`
      }
    }
  ],
  adapted: [
    {
      comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
      date: `2019-05-08T14:13:56.569Z`,
      id: 1,
      rating: 4,
      user: {
        avatarUrl: `img/1.png`,
        id: 4,
        isPro: false,
        name: `Max`
      }
    },
    {
      comment: `A quiet of Amsterdam.`,
      date: `2019-06-08T14:13:56.569Z`,
      id: 2,
      rating: 3,
      user: {
        avatarUrl: `img/2.png`,
        id: 5,
        isPro: false,
        name: `Vadim`
      }
    }
  ]
};
