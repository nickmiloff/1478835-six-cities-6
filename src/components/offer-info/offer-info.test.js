import React from 'react';
import {render} from '@testing-library/react';
import Test from '../test/test';

import OfferInfo from './offer-info';
import {cardsFull, comments} from '../../tests-mock';

describe(`OfferInfo component test`, () => {
  it(`OfferInfo component' should render correctly`, () => {
    const OFFER = cardsFull.adapted[0];
    const REVIEWS = comments.adapted;
    const MAP_PLACES = cardsFull.adapted;

    const {container} = render(
        <Test>
          <OfferInfo
            offer={OFFER}
            reviews={REVIEWS}
            mapPlaces={MAP_PLACES} />
        </Test>
    );

    expect(container).toMatchSnapshot();
  });
});
