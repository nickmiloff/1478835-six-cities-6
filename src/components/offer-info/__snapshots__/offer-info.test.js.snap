// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferInfo component test OfferInfo component' should render correctly 1`] = `
<div>
  <section
    class="property"
  >
    <div
      class="property__gallery-container container"
    >
      <div
        class="property__gallery"
      >
        <div
          class="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            class="property__image"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg"
          />
        </div>
        <div
          class="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            class="property__image"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/20.jpg"
          />
        </div>
        <div
          class="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            class="property__image"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/21.jpg"
          />
        </div>
        <div
          class="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            class="property__image"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/22.jpg"
          />
        </div>
        <div
          class="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            class="property__image"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/23.jpg"
          />
        </div>
        <div
          class="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            class="property__image"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/24.jpg"
          />
        </div>
      </div>
    </div>
    <div
      class="property__container container"
    >
      <div
        class="property__wrapper"
      >
        <div
          class="property__mark"
        >
          <span>
            Premium
          </span>
        </div>
        <div
          class="property__name-wrapper"
        >
          <h1
            class="property__name"
          >
            Tile House
          </h1>
          <button
            class="property__bookmark-button button"
            type="button"
          >
            <svg
              class="property__bookmark-icon"
              height="33"
              width="31"
            >
              <use
                xlink:href="#icon-bookmark"
              />
            </svg>
            <span
              class="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          class="property__rating rating"
        >
          <div
            class="property__stars rating__stars"
          >
            <span
              style="width: 40%;"
            />
            <span
              class="visually-hidden"
            >
              Rating
            </span>
          </div>
          <span
            class="property__rating-value rating__value"
          >
            2
          </span>
        </div>
        <ul
          class="property__features"
        >
          <li
            class="property__feature property__feature--entire"
          >
            house
          </li>
          <li
            class="property__feature property__feature--bedrooms"
          >
            2
             Bedrooms
          </li>
          <li
            class="property__feature property__feature--adults"
          >
            Max 
            9
             adults
          </li>
        </ul>
        <div
          class="property__price"
        >
          <b
            class="property__price-value"
          >
            €
            161
          </b>
          <span
            class="property__price-text"
          >
             night
          </span>
        </div>
        <div
          class="property__inside"
        >
          <h2
            class="property__inside-title"
          >
            What's inside
          </h2>
          <ul
            class="property__inside-list"
          >
            <li
              class="property__inside-item"
            >
              Laptop friendly workspace
            </li>
          </ul>
        </div>
        <div
          class="property__host"
        >
          <h2
            class="property__host-title"
          >
            Meet the host
          </h2>
          <div
            class="property__host-user user"
          >
            <div
              class="property__avatar-wrapper user__avatar-wrapper property__avatar-wrapper--pro"
            >
              <img
                alt="Host avatar"
                class="property__avatar user__avatar"
                height="74"
                src="img/avatar-angelina.jpg"
                width="74"
              />
            </div>
            <span
              class="property__user-name"
            >
              Angelina
            </span>
          </div>
          <div
            class="property__description"
          >
            <p
              class="property__text"
            >
              Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country. In our beautiful screened Pondhouse, you can gaze at the stars and listen to the sounds of nature from your cozy warm bed.
            </p>
          </div>
        </div>
        <section
          class="property__reviews reviews"
        >
          <h2
            class="reviews__title"
          >
            Reviews · 
            <span
              class="reviews__amount"
            >
              2
            </span>
          </h2>
          <ul
            class="reviews__list"
          >
            <li
              class="reviews__item"
            >
              <div
                class="reviews__user user"
              >
                <div
                  class="reviews__avatar-wrapper user__avatar-wrapper"
                >
                  <img
                    alt="Reviews avatar"
                    class="reviews__avatar user__avatar"
                    height="54"
                    src="img/1.png"
                    width="54"
                  />
                </div>
                <span
                  class="reviews__user-name"
                >
                  Max
                </span>
              </div>
              <div
                class="reviews__info"
              >
                <div
                  class="reviews__rating rating"
                >
                  <div
                    class="reviews__stars rating__stars"
                  >
                    <span
                      style="width: 80%;"
                    />
                    <span
                      class="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <p
                  class="reviews__text"
                >
                  A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.
                </p>
                <time
                  class="reviews__time"
                  datetime="2019-05-08T14:13:56.569Z"
                >
                  May 2019
                </time>
              </div>
            </li>
            <li
              class="reviews__item"
            >
              <div
                class="reviews__user user"
              >
                <div
                  class="reviews__avatar-wrapper user__avatar-wrapper"
                >
                  <img
                    alt="Reviews avatar"
                    class="reviews__avatar user__avatar"
                    height="54"
                    src="img/2.png"
                    width="54"
                  />
                </div>
                <span
                  class="reviews__user-name"
                >
                  Vadim
                </span>
              </div>
              <div
                class="reviews__info"
              >
                <div
                  class="reviews__rating rating"
                >
                  <div
                    class="reviews__stars rating__stars"
                  >
                    <span
                      style="width: 60%;"
                    />
                    <span
                      class="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <p
                  class="reviews__text"
                >
                  A quiet of Amsterdam.
                </p>
                <time
                  class="reviews__time"
                  datetime="2019-06-08T14:13:56.569Z"
                >
                  June 2019
                </time>
              </div>
            </li>
          </ul>
          
        </section>
      </div>
    </div>
    <section
      class="property__map map"
    >
      <div
        class="leaflet-container leaflet-grab leaflet-touch-drag"
        id="map"
        style="width: 100%; height: 100%; position: relative;"
        tabindex="0"
      >
        <div
          class="leaflet-pane leaflet-map-pane"
          style="left: 0px; top: 0px;"
        >
          <div
            class="leaflet-pane leaflet-tile-pane"
          >
            <div
              class="leaflet-layer "
              style="z-index: 1;"
            >
              <div
                class="leaflet-tile-container leaflet-zoom-animated"
                style="z-index: 18; left: 0px; top: 0px;"
              >
                <img
                  alt=""
                  class="leaflet-tile"
                  role="presentation"
                  src="https://a.basemaps.cartocdn.com/rastertiles/voyager/12/2074/1409.png"
                  style="width: 256px; height: 256px; left: -214px; top: -31px;"
                />
              </div>
            </div>
          </div>
          <div
            class="leaflet-pane leaflet-shadow-pane"
          />
          <div
            class="leaflet-pane leaflet-overlay-pane"
          />
          <div
            class="leaflet-pane leaflet-marker-pane"
          >
            <img
              alt=""
              class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
              src="img/pin-active.svg"
              style="margin-left: -15px; margin-top: -15px; width: 30px; height: 30px; left: 0px; top: 0px; z-index: 0;"
              tabindex="0"
            />
            <img
              alt=""
              class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
              src="img/pin.svg"
              style="margin-left: -15px; margin-top: -15px; width: 30px; height: 30px; left: 7402px; top: -16221px; z-index: -16221;"
              tabindex="0"
            />
            <img
              alt=""
              class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
              src="img/pin.svg"
              style="margin-left: -15px; margin-top: -15px; width: 30px; height: 30px; left: -24px; top: 85px; z-index: 85;"
              tabindex="0"
            />
            <img
              alt=""
              class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
              src="img/pin-active.svg"
              style="margin-left: -15px; margin-top: -15px; width: 30px; height: 30px; left: 0px; top: 0px; z-index: 0;"
              tabindex="0"
            />
          </div>
          <div
            class="leaflet-pane leaflet-tooltip-pane"
          />
          <div
            class="leaflet-pane leaflet-popup-pane"
          />
        </div>
        <div
          class="leaflet-control-container"
        >
          <div
            class="leaflet-top leaflet-left"
          />
          <div
            class="leaflet-top leaflet-right"
          />
          <div
            class="leaflet-bottom leaflet-left"
          />
          <div
            class="leaflet-bottom leaflet-right"
          >
            <div
              class="leaflet-control-attribution leaflet-control"
            >
              <a
                href="https://leafletjs.com"
                title="A JS library for interactive maps"
              >
                Leaflet
              </a>
               | © 
              <a
                href="https://www.openstreetmap.org/copyright"
              >
                OpenStreetMap
              </a>
               contributors © 
              <a
                href="https://carto.com/attributions"
              >
                CARTO
              </a>
            </div>
          </div>
        </div>
      </div>
    </section>
  </section>
</div>
`;
